packages/chevrotain/src/parse/cst/cst.ts

```diff
@@ -24,11 +24,45 @@ import { GAstVisitor } from "../grammar/gast/gast_visitor_public"
 import {
     CstChildrenDictionary,
     CstNode,
+    CstNodeLocation,
     IOptionallyNamedProduction,
     IProduction,
     IToken
 } from "../../../api"
 
+export function setNodeLocationOnlyOffset(
+    nodeLocation: CstNodeLocation,
+    locationInformation: any
+): void {
+    if (nodeLocation.startOffset > locationInformation.startOffset) {
+        nodeLocation.startOffset = locationInformation.startOffset
+    }
+}
+
+export function setNodeLocationOnlyStart(
+    nodeLocation: CstNodeLocation,
+    locationInformation: any
+): void {
+    if (nodeLocation.startOffset > locationInformation.startOffset) {
+        nodeLocation.startOffset = locationInformation.startOffset
+        nodeLocation.startColumn = locationInformation.startColumn
+        nodeLocation.startLine = locationInformation.startLine
+    }
+}
+
+export function setNodeLocationFull(
+    nodeLocation: CstNodeLocation,
+    locationInformation: any
+): void {
+    setNodeLocationOnlyStart(nodeLocation, locationInformation)
+
+    if (nodeLocation.endOffset < locationInformation.endOffset) {
```

Comment:
As mentioned before we are parsing left to right.
In this case it means we will always enter this condition (ignoring error recovery situations).